src/gevent_server.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+from gevent.server import StreamServer
+from gevent.monkey import patch_all
+import sys
+from server import parse_request, server_read, response_ok, server_response, response_error
+
+IP = '127.0.0.1'
+PORT = int(sys.argv[1])
+
+def server_handler(connection, address):
+    """Handles the gevent StreamServer for individual clients."""
+    try:
+        result, mime = parse_request(server_read(connection))
```

Comment:
notice how the call to `parse_request` here is returning the values that should be returned by `resolve_uri`.  This is going to make it much more difficult to test the `parse_uri` function because it is no longer possible to isolate the return values and errors it might produce from those that might be made by the nested call to `resolve_uri`.